app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt

```diff
@@ -0,0 +1,370 @@
+) {
+  DatePickerDialog(
+      onDismissRequest = { onDismissRequest() },
+      confirmButton = {
+        Button(
+            onClick = {
+              onDismissRequest()
+
+              val selectedDate =
+                  Calendar.getInstance().apply {
+                    timeInMillis = datePickerState.selectedDateMillis!!
+                  }
+              onDateSelected(selectedDate)
+            }) {
+              Text("OK")
+            }
+      }) {
+        DatePicker(
+            state = datePickerState,
+            // Add more customization as needed
+        )
+      }
+}
+
+@Composable
+fun TagChip(tag: String, onRemove: () -> Unit) {
+  Surface(
+      modifier = Modifier.padding(end = 8.dp),
+      shape = RoundedCornerShape(52.dp),
+      color = MaterialTheme.colorScheme.primary,
+      contentColor = MaterialTheme.colorScheme.onPrimary) {
+        Row(verticalAlignment = Alignment.CenterVertically) {
+          Text(
+              text = tag,
+              modifier = Modifier.padding(start = 16.dp),
+              style = MaterialTheme.typography.bodyMedium)
+          IconButton(onClick = onRemove) {
+            Icon(imageVector = Icons.Rounded.Close, contentDescription = "Remove tag")
+          }
+        }
+      }
+}
+
+@Composable
+fun FindEventLegend(text: String, imageVector: ImageVector, contentDescription: String) {
+  Row(
+      verticalAlignment = Alignment.CenterVertically,
+  ) {
+    Icon(imageVector = imageVector, contentDescription = contentDescription)
+    Text(
+        text = text,
+        modifier = Modifier.padding(8.dp),
+        style = MaterialTheme.typography.titleLarge,
+    )
+  }
+}
+
+@ExperimentalMaterial3Api
+@Composable
+fun FindEventMapScreen(
+    onBackIconClicked: () -> Unit,
+    locationHelper: LocationHelper = LocationHelper()
+) {
+  val scope = rememberCoroutineScope()
+  val scaffoldState = rememberBottomSheetScaffoldState()
+  val coroutineScope = rememberCoroutineScope()
+  var isMapInitialized by remember { mutableStateOf(false) }
+
+  val expandBottomSheet = { scope.launch { scaffoldState.bottomSheetState.expand() } }
+
+  val addMarker = { location: Location ->
+    coroutineScope.launch { locationHelper.addMarker(location) }
+  }
+
+  val removeMarker = { location: Location ->
+    coroutineScope.launch { locationHelper.removeMarker(location) }
+  }
+
+  addMarker(Location("Balelec", 46.519144, 6.566804))
+  addMarker(Location("AirSound", 46.559144, 6.566804))
+  addMarker(Location("Anniversaire Juan", 46.51644, 6.53804))
+
+  val systemUiPadding = WindowInsets.systemBars.asPaddingValues()
+
+  BottomSheetScaffold(
+      sheetContent = {
+        Box(Modifier.fillMaxWidth().height(128.dp), contentAlignment = Alignment.Center) {
+          Text("Swipe up to expand sheet")
+        }
+        Column(
+            Modifier.fillMaxWidth().padding(64.dp),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              Text("Sheet content")
```

Comment:
Use strings defined in resources for text, this allows for translations

Reply:
Done